src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/geo/OilPump.java

```diff
@@ -104,12 +135,12 @@ protected MachineRecipe findNextRecipe(BlockMenu inv) {
                         return recipe;
                     } else {
                         /*
-                         * Move the empty bucket to the output slot to prevent this
+                         * Set the oil pump on cooldown to prevent it
                          * from immediately starting all over again (to prevent lag)
+                         * We don't move the bucket into the output slots as it can cause the entire
+                         * system to be filled with buckets.
                          */
-                        ItemStack item = inv.getItemInSlot(slot).clone();
-                        inv.replaceExistingItem(slot, null);
-                        inv.pushItem(item, getOutputSlots());
+                        this.setOnResourceCooldown(b);
```

Comment:
@JustAHuman-xD as you can see here, this PR removes the bucket cycling already, which is what you're suggesting. It's already done.